src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -1291,6 +1286,12 @@ InstanceKlass* ClassLoader::load_class(Symbol* name, bool search_append_only, TR
 }
 
 #if INCLUDE_CDS
+char* ClassLoader::get_native_path(char* path) {
+  char* native_path = os::strdup(path);
+  assert(native_path != NULL, "sanity");
+  return os::native_path(native_path);
+}
+
```

Comment:
If I understand correctly, the reason to call `os::native_path()` is the Windows version of `os::same_files()` cannot handle the difference between "/" (from `src`) and "\\" (from `ent->name()`).

I think it's better to move the call to native_path inside the Windows implementation of `os::same_files()`. That way, we don't need to call `os::native_path()` in all the callers of `os::same_files()`.

This would also make it possible to use resource allocation inside `os::same_files()` to allocate the native path (instead of os::strdup).

Reply:
I've added `os::native_path()` calls in the Windows version of `os::same_files()`.
`os::strip()` and `os::free()` are still being used because ResourceMark may not be available during early VM startup.